### PR TITLE
Show commit information in error message

### PR DESCRIPTION
**Pull request by DileSoft**
image

### CODE REVIEW

File: lib/xtool/xerrhead.cpp [hunk 1]
```diff
@@ -47,28 +47,29 @@ void XErrorHandler::RTC(const char *file, unsigned int line, const char *expr)
...
     char time_buf[sizeof "2011-10-08T07:07:09Z"];
     strftime(time_buf, sizeof time_buf, "%Y-%m-%dT%H:%M:%SZ", gmtime(&now));
 
-	log_file.open(log_name.c_str(),std::ios::out|std::ios::app);
-	log_file<<time_buf<<" ";
-	log_file<<"Abort: "<<message<<" code:"<<code<<" val:"<<val<<std::endl;
+	stream<<time_buf<<" "<<std::endl;
+		stream<<GIT_BRANCH<<GITHUB_BRANCH<<" "<<GIT_COMMIT_HASH<<GITHUB_COMMIT_HASH<<std::endl<<std::endl;
+
+stream<<"Abort: "<<message<<" code:"<<code<<" val:"<<val<<std::endl;
```
Comment (stalkerg) [inline lib/xtool/xerrhead.cpp:59]: Something wrong with the indent here and in other parts of the changes.

File: CMakeLists.txt [hunk 2]
```diff
@@ -87,6 +87,30 @@ IF(WIN32)
 	)
 ENDIF(WIN32)
 
+# Get the current working branch
+IF(NOT($ENV{GITHUB_REF_NAME}))
```
Comment (stalkerg) [inline CMakeLists.txt:91]: It's ok but you should understand what Steam version build from the different repo.

Comment (DileSoft) [reply CMakeLists.txt:91]: I think, this is better than nothing, in this case we can ask you, what commit was this.

File: lib/xtool/xerrhead.cpp (hunk 1 above)
Comment (DileSoft) [reply lib/xtool/xerrhead.cpp]: fixed

**Comment by stalkerg** [comment]
Thanks! Let's try to work with it.

**Pull request merged**